Key/index.spec.ts

```diff
@@ -0,0 +1,91 @@
+		"DY5IiQHbIH83D4bvXq0X7qQoSBSNP7Dvv3HYuqMhf0DaegrlBuJllFVVq9qPVRnK" +
+		"xt1Il2HgxOBvbhOT+9in1BzA+YJ99UzC85O0Qz06A+CmtHEy4aZ2kj5hHjECgYEA" +
+		"mNS4+A8Fkss8Js1RieK2LniBxMgmYml3pfVLKGnzmng7H2+cwPLhPIzIuwytXywh" +
+		"2bzbsYEfYx3EoEVgMEpPhoarQnYPukrJO4gwE2o5Te6T5mJSZGlQJQj9q4ZB2Dfz" +
+		"et6INsK0oG8XVGXSpQvQh3RUYekCZQkBBFcpqWpbIEsCgYAnM3DQf3FJoSnXaMhr" +
+		"VBIovic5l0xFkEHskAjFTevO86Fsz1C2aSeRKSqGFoOQ0tmJzBEs1R6KqnHInicD" +
+		"TQrKhArgLXX4v3CddjfTRJkFWDbE/CkvKZNOrcf1nhaGCPspRJj2KUkj1Fhl9Cnc" +
+		"dn/RsYEONbwQSjIfMPkvxF+8HQ=="
+	const publicKey =
+		"MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAu1SU1LfVLPHCozMxH2Mo" +
+		"4lgOEePzNm0tRgeLezV6ffAt0gunVTLw7onLRnrq0/IzW7yWR7QkrmBL7jTKEn5u" +
+		"+qKhbwKfBstIs+bMY2Zkp18gnTxKLxoS2tFczGkPLPgizskuemMghRniWaoLcyeh" +
+		"kd3qqGElvW/VDL5AaWTg0nLVkjRo9z+40RQzuVaE8AkAFmxZzow3x+VJYKdjykkJ" +
+		"0iT9wCS0DRTXu269V264Vf/3jvredZiKRkgwlL9xNAwxXFg0x/XFw005UWVRIkdg" +
+		"cKWTjpBP2dPwVZ4WWC+9aGVd+Gyn1o0CLelf4rEjGoXbAAEgAqeGUxrcIlbjXfbc" +
+		"mwIDAQAB"
+	const creatable: issuefab.Key.Creatable = {
+		name: { first: "jessie", last: "doe" },
+		email: "jessie@example.com",
+		permissions: {
+			"*": {
+				org: true,
+				payment: { expense: true },
+				delegation: true,
+				app: {
+					view: true,
+				},
+			},
+			a1b2c3d4: {
+				org: true,
+				payment: { expense: true },
+				delegation: true,
+			},
+		},
+	}
+	const key: issuefab.Key = {
+		...creatable,
+		audience: "issuefab",
+		expires: "2023-08-19T00:25:25.000Z",
+		issued: "2023-08-18T12:08:21.000Z",
+		issuer: "userwidgets",
+		token:
+			"eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJ1c2Vyd2lkZ2V0cyIsImlhdCI6MTY5MjM2MTUyNSwiYXVkIjoiaXNzdWVmYWIiLCJleHAiOjE2OTI0MDQ3MjUsIm5hbSI6eyJmaXJzdCI6Implc3NpZSIsImxhc3QiOiJkb2UifSwic3ViIjoiamVzc2llQGV4YW1wbGUuY29tIiwicGVyIjoiKi5vcmcgKi5wYXltZW50LmV4cGVuc2UgKi5kZWxlZ2F0aW9uICouYXBwLnZpZXcgYTFiMmMzZDQub3JnIGExYjJjM2Q0LnBheW1lbnQuZXhwZW5zZSBhMWIyYzNkNC5kZWxlZ2F0aW9uIn0.irrOnQOKWlG5TcD1PxBCbiVFQYhMVbueKvtvFyBIy8ueEGftgvltCNhBfW3cx0AJunSP-f6rQwjPFU4huWQlo54UDOTXk7AX6igLDSThB44W-eRCUGHclid_67KIwsq7c5uVqWAbb4UMLILWHMg8ooYafKpeCtjlfLNNRTlYHduZz1dLQSoPqJR40Jpl0UmfKMGXXP-OXQH9cStKg2ZY9L76prhFcH1wTcCFp4-1GjpxI8Ewd-O_djpkRJ8OJCq07X7BckJAyOOo2dV5dOZW9Bb-0ncfKKNDmY2qnhw3jHWO8R5HF0yvkUF05edh_7uE57dPM3vqpspHkfoU_VvORQ",
+	}
+	it("is", () => {
+		expect(issuefab.Key.is(key)).toEqual(true)
+		expect(issuefab.Key.Creatable.type.get(key)).not.toEqual(undefined)
+		expect(issuefab.Key.type.get(key)).toEqual(key)
+	})
+	it("signing and verifying", async () => {
+		const issuer = userwidgets.User.Key.Issuer.create("userwidgets", "issuefab", publicKey, privateKey)
+		const token = await issuer.sign(creatable, now.getTime() / 1000)
+		if (token == undefined) {
+			expect(token).not.toEqual(undefined)
+			return
+		}
+		const verifier = issuefab.Key.Verifier.create(publicKey)
+		const key = await verifier.verify(token)
+		if (key == undefined) {
+			expect(key).not.toEqual(undefined)
+			return
+		}
```

Comment:
How does this work ?

Reply:
key is either undefined or a key. if its undefined the test should fail which is fixed with the expect (no if is necessary for that) but typescript will not understand that jest will stop executing after the failed expect so if key is undefined we return making typescript understand that after the if key is guaranteed to be a key and not undefined. 

So with this setup jest will error and typescript understand that execution will stop if that happens.